src/storage/engines/FirebaseStorageEngine.ts

```diff
@@ -371,7 +372,10 @@ export class FirebaseStorageEngine extends CloudStorageEngine {
     const revisitModesData = await getDoc(revisitModesDoc);
 
     if (revisitModesData.exists()) {
-      return revisitModesData.data() as Record<REVISIT_MODE, boolean>;
+      const modes = revisitModesData.data() as Record<string, boolean>;
+      const cleanedModes = cleanupModes(modes);
+      await setDoc(revisitModesDoc, cleanedModes);
```

Comment:
This is guaranteed to send a request every time we pull the modes. We should design this in a way to minimize calls to the DB. That is, we should add a check here and only do the cleanup if necessary